### PR TITLE
affine: add affine building from source

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/af/affine-bin/package.nix [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  pkgs,
+  electron,
+  commandLineArgs ? "",
+}:
+let
+  buildType = "stable";
+  # https://github.com/toeverything/AFFiNE/releases/tag/v0.18.1
+  version = "0.18.1";
+  rev = "8b066a4b398aace25a20508a8e3c1a381721971f";
+  sha256 = "TWwojG3lqQlQFX3BKoFjJ27a3T/SawXgNDO6fP6gW4k=";
```
[the remark this answers is not in the document]
Comment (xiaoxiangmoe) [reply pkgs/by-name/af/affine-bin/package.nix:18]: I'm sorry, but I'm not sure how to modify this. Can you give a more detailed example of the modification?

[the remark this answers is not in the document]
Comment (xiaoxiangmoe) [reply pkgs/by-name/af/affine-bin/package.nix:18]: I'm sorry, but I'm not sure how to modify this. Can you give a more detailed example of the modification?

**Comment by redyf** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358160`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>affine</li>
    <li>affine-bin</li>
  </ul>
</details>

**Review by redyf — APPROVED** [review]

**Comment by xiaoxiangmoe** [comment]
@wegank Is there any tutorial to fix this ci error?

**Comment by xiaoxiangmoe** [comment]
What should I do now?

**Review by ri-char — APPROVED** [review]

**Comment by wegank** [comment]
@ofborg eval

**Comment by qweered** [comment]
Do we want to keep affine-bin given that we have this now?

**Pull request merged**